src/libraries/Common/src/Interop/Unix/System.Native/Interop.ReadLink.cs

```diff
@@ -63,8 +72,33 @@ internal static partial class Sys
                 }
 
                 // Output buffer was too small, loop around again and try with a larger buffer.
-                outputBufferSize = buffer.Length * 2;
+                outputBufferSize = outputBufferSize * 2;
```

Comment:
This used the actual size of the previous buffer for reason. 

`ArrayPool.Rent` may give you several times larger buffer than what you have asked for. Using the actual size of the previous buffer can save multiple iterations of the buffer resizing.